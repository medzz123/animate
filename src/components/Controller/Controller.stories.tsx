// import { Meta, Story } from '@storybook/react/types-6-0';
// import React from 'react';

// import Controller from './Controller';

// export default {
//   title: 'Components/Controller',
//   component: Controller,
// } as Meta;

// const Template: Story = (storyArguments) => <Controller {...storyArguments} />;

// export const ControllerStory = Template.bind({});
