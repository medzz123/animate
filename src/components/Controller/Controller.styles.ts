import { tokens } from '@theme/tokens';
import styled from 'styled-components';

export const ControllerContainer = styled.div`
  overflow: scroll;
  height: ${tokens.sizes.full};
  padding: ${tokens.sizes[8]};
  min-width: ${tokens.sizes['2xs']};

  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */

  &::-webkit-scrollbar {
    display: none;
  }
`;
