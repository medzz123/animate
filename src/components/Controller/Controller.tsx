import Accordion from '@components/Accordion';
import AnimatableInput from '@components/AnimatableInput';
import Flex from '@components/Box/Flex';
import React, { FunctionComponent } from 'react';
import {
  FaArrowsAlt,
  FaBorderStyle,
  FaDiceD6,
  FaPalette,
  FaRulerCombined,
  FaSpellCheck,
} from 'react-icons/fa';

import { ControllerContainer } from './Controller.styles';

const Controller: FunctionComponent = () => {
  return (
    <ControllerContainer data-testid="controller">
      <Flex justifyContent="space-between">
        <h3>Red Box</h3>
        <h3>15%</h3>
      </Flex>
      <Accordion title="Transform" id="transform-properties" Icon={FaArrowsAlt}>
        <AnimatableInput label="Origin" placeholder="center center" />
        <AnimatableInput label="Translate" />
        <AnimatableInput label="Rotate" />
        <AnimatableInput label="Scale" />
        <AnimatableInput label="Skew" />
      </Accordion>
      <Accordion title="3D" id="3d-properties" Icon={FaDiceD6}>
        <AnimatableInput label="Perspective" />
        <AnimatableInput label="Translate3d" />
        <AnimatableInput label="Scale3d" />
        <AnimatableInput label="Rotate3d" />
      </Accordion>
      <Accordion
        title="Position"
        id="position-properties"
        Icon={FaRulerCombined}
      >
        <AnimatableInput label="Top" />
        <AnimatableInput label="Right" />
        <AnimatableInput label="Bottom" />
        <AnimatableInput label="Left" />
      </Accordion>
      <Accordion title="Color" id="color-properties" Icon={FaPalette}>
        <AnimatableInput label="Background" />
        <AnimatableInput label="Opacity" />
      </Accordion>
      <Accordion title="Text" id="text-properties" Icon={FaSpellCheck}>
        <AnimatableInput label="Color" />
        <AnimatableInput label="Font Size" />
      </Accordion>
      <Accordion title="Borders" id="border-properties" Icon={FaBorderStyle}>
        <AnimatableInput label="Border" />
        <AnimatableInput label="Radius" />
        <AnimatableInput label="Box Shadow" />
        <AnimatableInput label="Outline" />
      </Accordion>
    </ControllerContainer>
  );
};

export default Controller;
