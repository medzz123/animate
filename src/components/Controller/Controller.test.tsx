import { render } from '@test/testUtils';
import React from 'react';

import Controller from './Controller';

test('Controller Test', () => {
  const { container } = render(<Controller />);

  expect(container).toMatchSnapshot();
});
