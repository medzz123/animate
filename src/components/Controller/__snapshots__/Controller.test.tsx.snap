// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller Test 1`] = `
<div>
  <div
    class="sc-pFZIQ hiTkCY"
    data-testid="controller"
  >
    <div
      class="sc-fubCfw fhbzAc"
      data-testid="box"
    >
      <h3>
        Red Box
      </h3>
      <h3>
        15%
      </h3>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-2"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-1"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              Transform
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M352.201 425.775l-79.196 79.196c-9.373 9.373-24.568 9.373-33.941 0l-79.196-79.196c-15.119-15.119-4.411-40.971 16.971-40.97h51.162L228 284H127.196v51.162c0 21.382-25.851 32.09-40.971 16.971L7.029 272.937c-9.373-9.373-9.373-24.569 0-33.941L86.225 159.8c15.119-15.119 40.971-4.411 40.971 16.971V228H228V127.196h-51.23c-21.382 0-32.09-25.851-16.971-40.971l79.196-79.196c9.373-9.373 24.568-9.373 33.941 0l79.196 79.196c15.119 15.119 4.411 40.971-16.971 40.971h-51.162V228h100.804v-51.162c0-21.382 25.851-32.09 40.97-16.971l79.196 79.196c9.373 9.373 9.373 24.569 0 33.941L425.773 352.2c-15.119 15.119-40.971 4.411-40.97-16.971V284H284v100.804h51.23c21.382 0 32.09 25.851 16.971 40.971z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-1"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-2"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-4"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-3"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              3D
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M422.19 109.95L256.21 9.07c-19.91-12.1-44.52-12.1-64.43 0L25.81 109.95c-5.32 3.23-5.29 11.27.06 14.46L224 242.55l198.14-118.14c5.35-3.19 5.38-11.22.05-14.46zm13.84 44.63L240 271.46v223.82c0 12.88 13.39 20.91 24.05 14.43l152.16-92.48c19.68-11.96 31.79-33.94 31.79-57.7v-197.7c0-6.41-6.64-10.43-11.97-7.25zM0 161.83v197.7c0 23.77 12.11 45.74 31.79 57.7l152.16 92.47c10.67 6.48 24.05-1.54 24.05-14.43V271.46L11.97 154.58C6.64 151.4 0 155.42 0 161.83z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-3"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-4"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-6"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-5"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              Position
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M160 288h-56c-4.42 0-8-3.58-8-8v-16c0-4.42 3.58-8 8-8h56v-64h-56c-4.42 0-8-3.58-8-8v-16c0-4.42 3.58-8 8-8h56V96h-56c-4.42 0-8-3.58-8-8V72c0-4.42 3.58-8 8-8h56V32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v448c0 2.77.91 5.24 1.57 7.8L160 329.38V288zm320 64h-32v56c0 4.42-3.58 8-8 8h-16c-4.42 0-8-3.58-8-8v-56h-64v56c0 4.42-3.58 8-8 8h-16c-4.42 0-8-3.58-8-8v-56h-64v56c0 4.42-3.58 8-8 8h-16c-4.42 0-8-3.58-8-8v-56h-41.37L24.2 510.43c2.56.66 5.04 1.57 7.8 1.57h448c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-5"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-6"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-8"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-7"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              Color
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M204.3 5C104.9 24.4 24.8 104.3 5.2 203.4c-37 187 131.7 326.4 258.8 306.7 41.2-6.4 61.4-54.6 42.5-91.7-23.1-45.4 9.9-98.4 60.9-98.4h79.7c35.8 0 64.8-29.6 64.9-65.3C511.5 97.1 368.1-26.9 204.3 5zM96 320c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm32-128c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128-64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-7"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-8"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-10"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-9"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              Text
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M272 256h91.36c43.2 0 82-32.2 84.51-75.34a79.82 79.82 0 0 0-25.26-63.07 79.81 79.81 0 0 0 9.06-44.91C427.9 30.57 389.3 0 347 0h-75a16 16 0 0 0-16 16v224a16 16 0 0 0 16 16zm40-200h40a24 24 0 0 1 0 48h-40zm0 96h56a24 24 0 0 1 0 48h-56zM155.12 22.25A32 32 0 0 0 124.64 0H99.36a32 32 0 0 0-30.48 22.25L.59 235.73A16 16 0 0 0 16 256h24.93a16 16 0 0 0 15.42-11.73L68.29 208h87.42l11.94 36.27A16 16 0 0 0 183.07 256H208a16 16 0 0 0 15.42-20.27zM89.37 144L112 75.3l22.63 68.7zm482 132.48l-45.21-45.3a15.88 15.88 0 0 0-22.59 0l-151.5 151.5-55.41-55.5a15.88 15.88 0 0 0-22.59 0l-45.3 45.3a16 16 0 0 0 0 22.59l112 112.21a15.89 15.89 0 0 0 22.6 0l208-208.21a16 16 0 0 0-.02-22.59z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-9"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-10"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
    <div
      type="multiple"
    >
      <div
        data-state="closed"
      >
        <h3
          data-state="closed"
        >
          <button
            aria-controls="radix-id-0-12"
            aria-expanded="false"
            class="sc-bdfBwQ kdXUbr"
            data-state="closed"
            id="radix-id-0-11"
          >
            <svg
              aria-hidden="true"
              class="sc-eCssSg fCzzfS"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
              />
            </svg>
            <span
              class="sc-gsTCUz dcNvVw"
            >
              Borders
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240 416h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-96 0h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm192 0h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm96-192h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 96h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 96h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-288h-32a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-96H32A32 32 0 0 0 0 64v400a16 16 0 0 0 16 16h32a16 16 0 0 0 16-16V96h368a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
              />
            </svg>
            <div
              class="sc-dlfnbm dJABeR"
            />
          </button>
        </h3>
        <div
          aria-labelledby="radix-id-0-11"
          class="sc-hKgILt kElTmy"
          data-state="closed"
          hidden=""
          id="radix-id-0-12"
          role="region"
          style="--radix-accordion-panel-height: var(--radix-collapsible-content-height);"
        />
      </div>
    </div>
  </div>
</div>
`;
